js/src/logger.ts

```diff
@@ -3493,6 +3493,164 @@ export function traced<IsAsyncFlush extends boolean = true, R = void>(
   }
 }
 
+/**
+ * Check if a function is a sync generator function.
+ * @param fn The function to check.
+ * @returns True if the function is a sync generator function.
+ */
+function isGeneratorFunction(fn: any): boolean {
```

Comment:
May be better to define an actual generated method and to stringify that to get the actual expected string. Looks like a similar library does something similar: https://github.com/inspect-js/is-generator-function/blob/main/index.js#L25

Reply:
added a comment here, we don't need the level of backwards compat that the linked implementation uses. added some tests to verify as well